### PR TITLE
Set db-operator version on database CR during full reconcile

### PR DESCRIPTION
**Pull request by bobertrublik**
* Issue: https://github.com/db-operator/db-operator/issues/79

I had to add a namespace environment variable to allow db-operator to retrieve the namespace in which it runs in. Other than that I simply added a new field `version` to the `Database` CR.

### CODE REVIEW

File: charts/db-operator/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 type: application
 name: db-operator
-version: 1.23.0
+version: 1.23.1
```
Comment (allanger) [inline charts/db-operator/Chart.yaml:4]: Let's update the minor version, it doesn't seem like a patch to me

**Review by allanger — CHANGES_REQUESTED** [review]
One small thing

**Comment by allanger** [comment]
But also, CI has failed, so it would be nice to check what's wrong

File: charts/db-operator/Chart.yaml (hunk 1 above)
Comment (bobertrublik) [reply charts/db-operator/Chart.yaml]: fixed

**Comment by bobertrublik** [comment]
> But also, CI has failed, so it would be nice to check what's wrong

Well, it looks like it's working as expected. 😅

> 2024-04-25T18:41:01Z	ERROR	an error occurred during the reconciliation	{"controller": "database", "controllerGroup": "kinda.rocks", "controllerKind": "Database", "Database": {"name":"db-operator-mysql","namespace":"db-operator-whz3ob85lz"}, "namespace": "db-operator-whz3ob85lz", "name": "db-operator-mysql", "reconcileID": "5f2862b2-729f-4b6f-805b-994981395cad", "error": "Database.kinda.rocks \"db-operator-mysql\" is invalid: operatorVersion: Required value"}

Can I somehow insert another image to the CI?

**Comment by allanger** [comment]
Ah, sure, we need to release the db-operator versions first, sorry

**Comment by allanger** [comment]
@bobertrublik 
I guess we should add a note to the `notes.txt`, because it's kind of breaking for ones who don't install CRDs with the chart. Also, I would update the README. Could you please add the migration plan to both of them?

**Comment by bobertrublik** [comment]
Hey, I added a migration plan to the README but I don't fully understand how the NOTES.txt works. At the top there is the following text

> Breaking changes and migration paths:
> 
> {{ .Chart.Version }} is not breaking, you don't have to worry

but how does it know which version is breaking and which is not? 😅

**Comment by allanger** [comment]
> but how does it know which version is breaking and which is not? 😅

Just replace 
```
{{ .Chart.Version }} is not breaking, you don't have to worry
```

with something like 

```
{{ .Chart.Version }} is not breaking if CRDs are managed by helm, otherwise please read

... Migration plan ...
```

**Comment by bobertrublik** [comment]
I hope this looks fine for you. 🙂

**Review by allanger — APPROVED** [review]

**Pull request merged**